odata-csdl/12 Action and Function.md

```diff
@@ -79,38 +78,39 @@ MUST be unique within its schema.
 
 Functions MAY be [composable](#ComposableFunction).
 
-The function MUST specify a [return type](#ReturnType) which MUST be a
+## ##subsec Function Overloads
+
+A function consists of one or more function overloads, all sharing the same function name.
+
+A function overload MUST specify a [return type](#ReturnType) which MUST be a
 primitive, entity or complex type, or a collection of primitive, entity
 or complex types in scope.
 
-A function MAY define [parameters](#Parameter) used during the execution
-of the function.
+A function overload MAY define [parameters](#Parameter) used during the execution
+of the function overload.
 
-## ##subsec Function Overloads
 
-[Bound](#BoundorUnboundActionorFunctionOverloads) functions support
-overloading (multiple functions having the same name within the same
-schema) subject to the following rules:
+[Bound](#BoundorUnboundActionorFunctionOverloads) function overloads are
+subject to the following rules:
 - The combination of function name,
 binding parameter type, and unordered set of non-binding parameter names
 MUST be unique within a schema.
 - The combination of function name,
 binding parameter type, and ordered set of parameter types MUST be
 unique within a schema.
-- All bound functions with the same
+- All bound function overloads with the same
 function name and binding parameter type within a schema MUST specify
-the same return type.
+the same [return type](#ReturnType) with identical nullability and type facets.
 
-[Unbound](#BoundorUnboundActionorFunctionOverloads) functions support
-overloading subject to the following rules:
+[Unbound](#BoundorUnboundActionorFunctionOverloads) function overloads are
+subject to the following rules:
 - The combination of function name and
 unordered set of parameter names MUST be unique within a schema.
 - The combination of function name and
 ordered set of parameter types MUST be unique within a schema.
-- All unbound functions with the same
-function name within a schema MUST specify the same return type.
-
-An unbound function MAY have the same name as a bound function.
+- All unbound function overloads with the same
+function name within a schema MUST specify
+the same [return type](#ReturnType) with identical nullability and type facets.
 
 Note that [type definitions](#TypeDefinition) can be used to
 disambiguate overloads for both bound and unbound functions, even if
```

Comment:
1. Why "both bound and unbound"? Unbound function overloads can be distinguished by function name and parameter names, without looking at types.
2. unbound functions -> unbound function overloads
3. Does this also work for actions?

Reply:
The second rule "The combination of function name and ordered set of parameter types MUST be unique within a schema" ignores the parameter _names_, so overloads
~~~
function forecastFor(year: Edm.String)
function forecastFor(quarter: Edm.String)
~~~
would not be allowed as both have the same type. What would be allowed is
~~~
function forecastFor(year: My.Year)
function forecastFor(quarter: My.Quarter)

typedef Year = Edm.String
typedef Quarter = Edm.String
~~~

This is not necessary for actions as overload resolution is based exclusively on the binding parameter _type_.

Reply:
What's the problem with differently named parameters that have the same type? The parameter names always appear in the URL.

Reply:
@mikepizzo Was this a restriction in the .NET platform?

Reply:
This was to support languages that resolved overloads by parameter types.